inference_perf/client/storage/base.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC
+from typing import List
+
+from pydantic import BaseModel
+from datetime import datetime
+
+class StorageConfig(BaseModel):
+    pass
+
+class StorageClient(ABC):
+    """Abstract base class for cloud storage clients."""
+    def __init__(self) -> None:
+        timestamp = datetime.now().strftime("%Y%m%d-%H%M%S")
+        self.path = f"reports-{timestamp}"
+        print(f"Reports will be stored at: {self.path}")
+
+    def push_report(self, reports: List[str]) -> None:
```

Comment:
`save ` is a more generic term with storage solutions and appropriate for the abstract class.

Further, we should allow reportgen to pass a list of fileObjects (with filename and content). The storage client should be agnostic of reporting process and should only deal with storing of such file objects at the path. 
```suggestion
    def save_report(self, reports: List[ReportFile]) -> None:
```